pages/webview/content.py

```diff
@@ -2,82 +2,380 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import re
+
+from functools import wraps
+
 from selenium.webdriver.common.by import By
+from selenium.common.exceptions import StaleElementReferenceException
 
 from pages.webview.base import Page
 from regions.webview.base import Region
+from regions.webview.content_item import ContentItem
 
 
 class Content(Page):
-    _title_locator = (By.CLASS_NAME, 'large-header')
+    # An at sign, then one or more non-colon chars, followed by a forward slash
+    _url_regex = re.compile('@[^:]+/')
+    _section_title_div_locator = (By.CSS_SELECTOR, '#main-content div.media-header div.title')
+    _chapter_section_span_locator = (By.CSS_SELECTOR, 'span.title-chapter')
     _ncy_locator = (By.CLASS_NAME, 'not-converted-yet')
 
     @property
     def loaded(self):
-        return '@' in self.driver.current_url
+        return self._url_regex.search(self.driver.current_url)
+
+    @property
+    def content_header(self):
+        return self.ContentHeader(self)
+
+    @property
+    def header_nav(self):
+        return self.content_header.nav
 
     @property
     def title(self):
-        return self.find_element(*self._title_locator).text
+        return self.content_header.title
+
+    @property
+    def share(self):
+        return self.content_header.share
+
+    # Retries StaleElementReferenceExceptions up to n-1 times (default n=3)
+    # Decorator with optional argument based on: https://stackoverflow.com/a/3931903
+    def retry_stale_element_reference_exception(method_or_max_tries):
```

Comment:
This seems quite useful! We may need to place this somewhere other than in a page. How about we place it in utils.py?

Reply:
I'll move it, but I'd rather not have this in too many places.
Also, what should I do about the other version of this that is in the Get This Book! region (in the Get This Book! PR)